trade-finance-manager-ui/server/controllers/case/amendments/amendmentIsUsingFacilityEndDate.test.js

```diff
@@ -0,0 +1,251 @@
+import { isTfmFacilityEndDateFeatureFlagEnabled, TEAM_IDS } from '@ukef/dtfs2-common';
+import api from '../../../api';
+import { mockRes } from '../../../test-mocks';
+import {
+  MOCK_AMENDMENT_COVERENDDATE_CHANGE,
+  MOCK_AMENDMENT_FACILITYVALUE_AND_COVERENDDATE_CHANGE,
+  MOCK_AMENDMENT_FACILITYVALUE_CHANGE,
+} from '../../../test-mocks/amendment-test-mocks';
+import { getAmendmentIsUsingFacilityEndDate, postAmendmentIsUsingFacilityEndDate } from './amendmentIsUsingFacilityEndDate.controller';
+import CONSTANTS from '../../../constants';
+
+const res = mockRes();
+
+const user = {
+  _id: '12345678',
+  username: 'testUser',
+  firstName: 'Joe',
+  lastName: 'Bloggs',
+  teams: [TEAM_IDS.PIM],
+  email: 'test@localhost',
+};
+
+const session = { user, userToken: 'mockToken' };
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+jest.mock('@ukef/dtfs2-common', () => ({
+  ...jest.requireActual('@ukef/dtfs2-common'),
+  isTfmFacilityEndDateFeatureFlagEnabled: jest.fn(),
+}));
+
+describe('GET getAmendIsUsingFacilityEndDate', () => {
+  describe('TFM Facility end date feature flag disabled', () => {
+    beforeAll(() => {
+      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(false);
+    });
+
+    afterAll(() => {
+      jest.resetAllMocks();
+    });
+
+    it('should redirect to the amendment options selection page', async () => {
+      api.getAmendmentById = () => Promise.resolve({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE });
+
```

Comment:
I think I'd prefer something like

```suggestion
      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE });
```

with

```
api.getAmendmentById = jest.fn();
```
somewhere earlier in the file

and then we can use `jest.resetAllMocks();` in beforeEach statements to make sure nothing is accidentally getting carried over between tests

Reply:
Ok good shout, I've changed that in this file and in the `amendCoverEndDate.test.js` file too, have a look and check it now looks ok